src/Infrastructure/Stize.Infrastructure.Azure/Sql/SqlDatabaseExtensions.cs

```diff
@@ -67,36 +66,140 @@ public static SqlDatabaseBuilder ElasticPoolId(this SqlDatabaseBuilder builder,
         }
 
         /// <summary>
-        /// Restores the database from an existing one
+        /// Restores a deleted database using the database's original resource ID.
         /// </summary>
         /// <param name="builder"></param>
-        /// <param name="databaseId"></param>
+        /// <param name="databaseId">Original Resource ID of the database to restore from.</param>
+        /// <param name="deletionDate">Deletion date of the database in (ISO 8601 format).</param>
+        /// <param name="pointInTime">Point in time that you want to restore from (ISO 8601 format). Only specify to restore from an earlier point in time.</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsRestore(this SqlDatabaseBuilder builder, Input<string> databaseId, Input<string> deletionDate, 
+            Input<string> pointInTime = null)
+        {
+            /* Same comments SetAsRestore() method. 
+             * This method seems redundant as you don't need to set SourceDatabaseDeletionDate to restore a deleted database - the RestorableDroppedDatabaseId will suffice.
+             * "If sourceDatabaseId is the database�s original resource ID, then sourceDatabaseDeletionDate must be specified." - https://www.pulumi.com/docs/reference/pkg/azure-native/sql/database/#createmode_csharp
+             * ^ documentation states to assign SourceDatabaseId with the original resource Id of the database - though, CreateMode.Restore requires RestorableDroppedDatabaseId.
+            */
+            builder.Arguments.CreateMode = CreateMode.Restore;
+            builder.Arguments.RestorableDroppedDatabaseId = databaseId;
+            builder.Arguments.SourceDatabaseDeletionDate = deletionDate;
+            builder.Arguments.RestorePointInTime = pointInTime ?? null;
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets this database to be a secondary replica of an existing database, using the existing primary database's resource ID.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Resource ID of the primary database</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsSecondary(this SqlDatabaseBuilder builder, Input<string> databaseId)
+        {
+            builder.Arguments.CreateMode = CreateMode.Secondary;
+            builder.Arguments.SourceDatabaseId = databaseId;
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets the database to be a copy of an existing database
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Resource ID of the database to copy</param>
         /// <returns></returns>
-        public static SqlDatabaseBuilder RestoreFrom(this SqlDatabaseBuilder builder, Input<string> databaseId)
+        public static SqlDatabaseBuilder SetAsCopy(this SqlDatabaseBuilder builder, Input<string> databaseId)
```

Comment:
Should we rename this to "CreateAsCopy"? I feel the current name is a bit confusing.

Reply:
Why did we change "RestoreFrom" as the name? RestoreFrom.. RecoverFrom are nicely fluent.

Reply:
One other thought I had - should the database ID be passed in as the argument or the database object itself and then get the ID in the method? Passing in the db object might affect the performance a little bit but the method could read "RestoreFrom(database)"  -> just a thought